web/src/sections/actions/MCPActionCard.tsx

```diff
@@ -151,74 +160,103 @@ export default function MCPActionCard({
-            description={description}
-            icon={icon}
-            status={status}
-            onEdit={onEdit}
-          />
-
-          {/* Action Buttons */}
-          <Actions
-            status={status}
-            serverName={title}
-            onDisconnect={onDisconnect}
-            onManage={onManage}
-            onAuthenticate={onAuthenticate}
-            onReconnect={onReconnect}
-            onDelete={onDelete}
-            toolCount={toolCount}
-            isToolsExpanded={isToolsExpanded}
-            onToggleTools={handleToggleTools}
-          />
-        </div>
-
-        {/* Tools Section (Only when expanded) */}
-        {isToolsExpanded && (
-          <ToolsSection
-            onRefresh={() => onRefreshTools?.(serverId, mutate)}
-            onDisableAll={() => {
-              const toolIds = tools.map((tool) => parseInt(tool.id));
-              onDisableAllTools?.(serverId, toolIds, mutate);
-            }}
-            onFold={handleFold}
-            searchQuery={searchQuery}
-            onSearchQueryChange={setSearchQuery}
-          />
-        )}
-      </div>
-
-      {/* Tools List - Only render when expanded */}
-      {isToolsExpanded && (
-        <div className="animate-in fade-in slide-in-from-top-2 duration-300 p-2 border-t border-border-01">
-          <ToolsList
-            tools={filteredTools}
-            searchQuery={searchQuery}
-            onToolToggle={(toolId, enabled) =>
-              onToolToggle?.(serverId, toolId, enabled, mutate)
-            }
-            isFetching={server.status === MCPServerStatus.FETCHING_TOOLS}
-            onRetry={() => mutate()}
-          />
-        </div>
+    <>
+      <ActionCard
+        title={title}
+        description={description}
+        icon={icon}
+        status={status}
+        actions={actionsComponent}
+        onEdit={onEdit}
+        onRename={handleRename}
+        isExpanded={isToolsExpanded}
+        onExpandedChange={setIsToolsExpanded}
+        enableSearch={true}
+        searchQuery={searchQuery}
+        onSearchQueryChange={setSearchQuery}
+        onRefresh={() => onRefreshTools?.(serverId, mutate)}
+        onDisableAll={() => {
+          const toolIds = tools.map((tool) => parseInt(tool.id));
+          onDisableAllTools?.(serverId, toolIds, mutate);
+        }}
+        onFold={handleFold}
+        className={className}
+        ariaLabel={`${title} MCP server card`}
+      >
+        <ToolsList
+          isFetching={server.status === MCPServerStatus.FETCHING_TOOLS}
+          onRetry={() => mutate()}
+          totalCount={tools.length}
+          enabledCount={tools.filter((tool) => tool.isEnabled).length}
+          showOnlyEnabled={showOnlyEnabled}
+          onToggleShowOnlyEnabled={handleToggleShowOnlyEnabled}
+          isEmpty={filteredTools.length === 0}
+          searchQuery={searchQuery}
+          emptyMessage="No tools available"
+          emptySearchMessage="No tools found"
```

Comment:
Do these need to be passed in as parameters? Can `ToolsList` just have these hardcoded inside?